src/js/components/FileInput/FileInput.js

```diff
@@ -130,133 +130,206 @@ const FileInput = forwardRef(
+        message = 'Drop files here or' || messages.dropPromptMultiple;
+      if (!multiple) {
+        message = 'Drop file here or' || messages.dropPrompt;
+      }
+    } else message = `${files.length} items`;
+
     return (
-      <Keyboard
-        onSpace={event => {
-          if (controlRef.current === event.target) inputRef.current.click();
-        }}
-        onEnter={event => {
-          if (controlRef.current === event.target) inputRef.current.click();
-        }}
+      <ContentsBox
+        tabIndex={-1}
+        theme={theme}
+        disabled={disabled}
+        background={mergeTheme('background', 'color')}
+        border={mergeTheme('border', 'side')}
+        margin={mergeTheme('margin')}
+        pad={mergeTheme('pad')}
+        round={mergeTheme('round', 'size')}
+        align={files.length ? 'stretch' : 'center'}
+        justify="center"
+        hover={hover}
+        onMouseOver={() => setHover(true)}
+        onMouseOut={() => setHover(false)}
+        dragOver={dragOver}
       >
-        <ContentsBox
-          ref={controlRef}
-          tabIndex={!plain ? 0 : undefined}
-          theme={theme}
-          disabled={disabled}
-          background={mergeTheme('background', 'color')}
-          border={mergeTheme('border', 'side')}
-          margin={mergeTheme('margin')}
-          pad={mergeTheme('pad')}
-          round={mergeTheme('round', 'size')}
-          align={files.length ? 'stretch' : 'center'}
-          justify="center"
-          hover={hover}
-          onMouseOver={() => setHover(true)}
-          onMouseOut={() => setHover(false)}
-          dragOver={dragOver}
-        >
-          {files.length > aggregateThreshold && (
-            <Box direction="row" align="center" justify="between">
-              <Label {...theme.fileInput.label}>
-                {files.length} {messages.files || 'files'}
-              </Label>
+        {(!files.length || files.length > 1) && (
+          <Box direction="row" justify="between">
+            <Message {...theme.fileInput.message}>{message}</Message>
+            <Keyboard
+              onSpace={event => {
+                if (controlRef.current === event.target)
+                  inputRef.current.click();
+              }}
+              onEnter={event => {
+                if (controlRef.current === event.target)
+                  inputRef.current.click();
+              }}
+            >
+              {theme.fileInput.button ? (
+                <Button
+                  ref={controlRef}
+                  kind={theme.fileInput.button}
+                  label={messages.browse || 'browse'}
+                  onClick={() => {
+                    inputRef.current.click();
+                    inputRef.current.focus();
+                  }}
+                  {...rest}
+                />
+              ) : (
+                <Anchor
+                  alignSelf="center"
+                  ref={controlRef}
+                  margin="small"
```

Comment:
Since `anchor` does not have `pad` I added `margin="small"` wondering if we should add a `theme.fileinput.anchor.margin` that will take care of this

Reply:
I think it's fair to just decide this and not add anything to the theme for now -- would be easy enough to add in future if people needed it